code/modules/research/rdconsole.dm

```diff
@@ -147,471 +179,665 @@ won't update every console in existence) but it's more of a hassle to do. Also,
 		linked_imprinter = null
 	..()
 
-
 /obj/machinery/computer/rdconsole/emag_act(mob/user)
-	if(emagged)
-		return
-	playsound(src, "sparks", 75, 1)
-	emagged = TRUE
-	to_chat(user, "<span class='notice'>You disable the security protocols</span>")
+	if(!emagged)
+		to_chat(user, "<span class='notice'>You disable the security protocols</span>")
+		playsound(src, "sparks", 75, 1)
+	return ..()
 
-/obj/machinery/computer/rdconsole/Topic(href, href_list)
-	if(..())
+/obj/machinery/computer/rdconsole/proc/list_categories(list/categories, menu_num as num)
+	if(!categories)
 		return
 
-	add_fingerprint(usr)
+	var/line_length = 1
+	var/list/l = "<table style='width:100%' align='center'><tr>"
 
-	usr.set_machine(src)
-	if(href_list["disk_slot"])
-		disk_slot_selected = text2num(href_list["disk_slot"])
+	for(var/C in categories)
+		if(line_length > 2)
+			l += "</tr><tr>"

```

Comment:
I'm confused what are you trying to achieve here.

Reply:
its from the old code.